benchmarks/benchmarks/analysis/distances.py

```diff
@@ -11,120 +17,231 @@
         self.coords_2 = np.random.random_sample((num_atoms, 3)).astype(np.float32)
         self.allocated_array_2D = np.empty((num_atoms, num_atoms),
-                                            dtype=np.float64)
+                                           dtype=np.float64)
         self.array_shape_1D = int(num_atoms * (num_atoms - 1) / 2.)
         self.allocated_array_1D = np.empty(self.array_shape_1D,
                                            dtype=np.float64)
+
+    def time_between(self, num_atoms):
+        """Benchmark determination of subgroup
+        of atomgroup that is within a specific
+        distance of two other atomgroups.
+        """
+        distances.between(group=self.ag3,
+                          A=self.ag1,
+                          B=self.ag2,
+                          distance=15.0)
+
+
+class DistancesBench(object):
+    """Benchmarks for MDAnalysis.analysis.distances
+    functions. Excluding contact matrices.
+    """
+    params = ([10, 100, 1000, 10000], [None, 'orthogonal', 'triclinic'])
+    param_names = ['num_atoms', 'pbc_type']
+
+    def setup(self, num_atoms, pbc_type):
+
         self.u = MDAnalysis.Universe(GRO)
+
+        if pbc_type is None:
+            self.box_dims = None
+
+        elif pbc_type == 'orthogonal':
+            box_shape = self.u.dimensions
+            basis_vectors = mdamath.triclinic_vectors(box_shape)
+
+            # this will calculate the furthest point from the origin on
+            # the triclinic unit cell.
+            furthest_point = np.sum(basis_vectors, axis=0)
+            # get full size of cube and also deal with certain distance
+            # functions requiring float32
+            orthogonal_box_size = np.float32(np.max(furthest_point))
+            angle = np.float32(90.0)
+
+            # make dummy cubic box, must be np.array, distance functions do
+            # not accept list
+            self.box_dims = np.array([orthogonal_box_size,
+                                      orthogonal_box_size,
+                                      orthogonal_box_size,
+                                      angle, angle, angle])
+
+        elif pbc_type == 'triclinic':
+            self.box_dims = self.u.dimensions
+
         self.ag1 = self.u.atoms[:num_atoms]
         self.ag2 = self.u.atoms[num_atoms: 2 * num_atoms]
         self.ag3 = self.u.atoms[-num_atoms:]
 
-    def time_distance_array(self, num_atoms):
+        np.random.seed(17809)
+        self.coords_1 = np.random.random_sample((num_atoms, 3)).astype(np.float32)
+        np.random.seed(9008716)
+        self.coords_2 = np.random.random_sample((num_atoms, 3)).astype(np.float32)
+        self.allocated_array_2D = np.empty((num_atoms, num_atoms),
+                                           dtype=np.float64)
+        self.array_shape_1D = int(num_atoms * (num_atoms - 1) / 2.)
+        self.allocated_array_1D = np.empty(self.array_shape_1D,
+                                           dtype=np.float64)
```

Comment:
If we're generating synthetic coordinates (which isn't a bad idea) then we can probably not actually use an mda.Universe above.

Reply:
Got a quick link for how to easily manufacture dummy universe? A few functions require atom groups.

Reply:
You mentioning this made me play more around with it. The failiures in #2794 seems to be due to the fact that we use synthetic coordinates for certain functions.

When we draw samples from `np.random.random_samples` to generate synthetic coordinates we only get numbers between 0 and 1. 

'MDanalysis.analysis.distances.contacts_matrix' relies on a cutoff distance (default=15). So the benchmark was drawing edges between every atom. The 10 000 particles were causing memory overflow, not a time out issue. 

I can run the benchmark with 30 000 atoms on my 16GB laptop using realistic positions in the data files. The performance of this function seems to be very dependent on data that looks 'real'.